plugins/check_mysql_slave.c

```diff
@@ -0,0 +1,541 @@
+/*****************************************************************************
+*
+* Monitoring check_mysql plugin
+*
+* License: GPL
+* Copyright (c) 1999 Didi Rieder (adrieder@sbox.tu-graz.ac.at)
+* Copyright (c) 2000 Karl DeBisschop (kdebisschop@users.sourceforge.net)
+* Copyright (c) 1999-2011 Monitoring Plugins Development Team
+* Copyright (c) 2016 Zalora South East Asia Pte. Ltd
+*
+* Description:
+*
+* This file contains the check_mysql_slave plugin
+*
+* This program tests MySQL/MariaDB slaves
+*
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+*
+*****************************************************************************/
+
+const char *progname = "check_mysql_slave";
+const char *copyright = "1999-2016";
+const char *email = "devel@monitoring-plugins.org";
+
+#define SLAVERESULTSIZE 256
+
+#include "common.h"
+#include "utils.h"
+#include "utils_base.h"
+#include "netutils.h"
+
+#include <mysql.h>
+#include <mysqld_error.h>
+#include <errmsg.h>
+
+char *db_user = NULL;
+char *db_host = NULL;
+char *db_socket = NULL;
+char *db_pass = NULL;
+char *ca_cert = NULL;
+char *ca_dir = NULL;
+char *cert = NULL;
+char *key = NULL;
+char *ciphers = NULL;
+bool ssl = false;
+char *opt_file = NULL;
+char *opt_group = NULL;
+unsigned int db_port = MYSQL_PORT;
```

Comment:
Please don't reference MYSQL_PORT.  Instead use a 0 to have the library fall back to defaults or what is specified in the configuration files it will read.

Reply:
Good point. These days I'm inclined to purely rely on configuration files when connecting to MySQL servers.